### DISCUSSION TITLE
How to remove '/' from keys in a .yaml file if the '/' is the first character

### DISCUSSION THREAD

**Discussion by williamj20** [mikefarah/yq#1560] [Q&A] [answered]
Hi, I have a bunch of paths in a .yaml file which start with '/'. For example: '/api/path'. I would like to remove all instances of these paths which start with '/', so that they become 'api/path'. How can I achieve this?

**Comment by mikefarah** [comment, marked as answer]
You can use the `sub` operator (see https://mikefarah.gitbook.io/yq/operators/string-operators#sub-regex-replacement)  like this:

sample file (next time please provide one)
```yaml
- /a/path
- something/else
```

```
yq '.[] |= sub("^/", "")' file.yaml
```

output:
```yaml
- a/path
- something/else
```

Explanation:
- `.[]` will match all the elements in the array
- `|=` will update each element, with respect to itself
- `sub("^/", "")` matches a leading `/` and replaces it with an empty string
